addons/product_expiry/models/production_lot.py

```diff
@@ -59,17 +59,18 @@ def _onchange_product(self):
     @api.model
     def _alert_date_exceeded(self):
         # if the alert_date is in the past and the lot is linked to an internal quant
-        # log a next activity on the next.production.lot 
+        # log a next activity on the next.production.lot
         alert_lot_ids = self.env['stock.production.lot'].search([('alert_date', '<=', fields.Date.today())])
         mail_activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached').id
+        # VFE FIXME will crash if user deletes activity type...
```

Comment:
Indeed, if it is deleted then it should not crash; filter on lot will have to be skipped in that case I think. It will probably generate too much activities, not sure if it is called regularly or not. To check more in details.